src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java

```diff
@@ -41,55 +41,59 @@
 
         if (!executeSearchQueryFrom.equals(configRun)) {
 
             searchQuery = searchQuery + configRun;
 
-            log.info("{}:: searchQuery:: will execute from::DB job run value::", loggingComponentName, searchQuery);
+            log.info("{}:: searchQuery:: will execute from::DB job run value::{}" + searchQuery, loggingComponentName);
 
-        } else if (null != syncJobRepository.findFirstByStatusOrderByAuditTsDesc(SUCCESS)) {
+        } else if (null != profileSyncAuditRepository.findFirstBySchedulerStatusOrderBySchedulerEndTimeDesc(SUCCESS)) {
 
-            SyncJobAudit auditjob = syncJobRepository.findFirstByStatusOrderByAuditTsDesc(SUCCESS);
-            searchQuery = searchQuery + getLastBatchFailureTimeInHours(auditjob.getAuditTs());
+            ProfileSyncAudit syncAuditDtl  = profileSyncAuditRepository.findFirstBySchedulerStatusOrderBySchedulerEndTimeDesc(SUCCESS);
+            searchQuery = searchQuery + getLastBatchFailureTimeInHours(syncAuditDtl.getSchedulerEndTime());
 
-            log.info("{}::  SearchQuery::executing from last success ::", loggingComponentName, searchQuery);
+            log.info("{}::  SearchQuery::executing from last success ::{}" + searchQuery, loggingComponentName);
         }
 
         try {
-
-            profileSyncService.updateUserProfileFeed(searchQuery);
-            SyncJobAudit syncJobAudit = new SyncJobAudit(201, SUCCESS, Source.SYNC);
-            syncJobRepository.save(syncJobAudit);
+            syncAudit = profileSyncService.updateUserProfileFeed(searchQuery, syncAudit);
+            if (StringUtils.isEmpty(syncAudit.getSchedulerStatus())) {
+                syncAudit.setSchedulerStatus(SUCCESS);
+            }
+            syncAudit.setSchedulerStartTime(startTime);
+            //updating same sync update with status and start time
+            profileSyncAuditRepository.save(syncAudit);
 
             // setting the value to run next job for from
             if (!executeSearchQueryFrom.equals(configRun)) {
                 syncJobConfig.setConfigRun(executeSearchQueryFrom);
                 syncConfigRepository.save(syncJobConfig);
             }
-
-
+            log.info("{}::Sync batch job executed successfully::{}", loggingComponentName);
 
         } catch (UserProfileSyncException e) {
-            log.info("{}:: Sync Batch Job Failed::", loggingComponentName, e.getErrorMessage());
-            SyncJobAudit syncJobAudit = new SyncJobAudit(500, "fail", Source.SYNC);
-            syncJobRepository.save(syncJobAudit);
+            log.error("{}::Sync Batch Job Failed::{}", loggingComponentName, e.getErrorMessage());
+            syncAudit.setSchedulerStatus("fail");
+            syncAudit.setSchedulerStartTime(startTime);
+            profileSyncAuditRepository.save(syncAudit);
 
         }
     }
 
-    private String getLastBatchFailureTimeInHours(LocalDateTime lastSuccessBatch) {
+    public String getLastBatchFailureTimeInHours(LocalDateTime lastSuccessBatch) {
```

Comment:
the method name is wrong. this method is to calculate the time difference from last successful run and not for batch failure

Reply:
done